DashAISoftware/DashAI
# Initial view of results tab

# Summary


Initial view to visualize information about the runs using a dynamic table that changes its content according to the selected experiment

## Type of change



- Front end new feature.

## Changes


- A permanent drawer is added to select an experiment.
- A table is added to display the runs associated with an experiment.
- Added support for the table to display content based on the route
- Added `api/run.ts` to get the runs data from the api.

## How to Test


1. Go to `DashAI/DashAI/front` and run `yarn json-server`
2. in another terminal in the same folder of step 1 run `yarn start`
4. Navigate to the results tab
5.  Click on an experiment and the table should show the information of the associated runs
6. In the upper part of the table you can configure the columns displayed, and other settings


## Screenshots


[results.webm](https://github.com/DashAISoftware/DashAI/assets/53657198/35935062-329e-4e95-89d7-a8b17fa0c2f8)


## Notes


- This PR depends on #96 
- `RunsTable.jsx` uses an [experimental DataGrid feature](https://mui.com/x/react-data-grid/column-groups/) to group columns, if needed it can be easily removed.
